### PR TITLE
Prefer IndexLambda for elementwise expressions

### PR DESCRIPTION
**Pull request by inducer**
@kaushikcfd How do you feel about this?

x-ref: #17

### CODE REVIEW

File: doc/design.rst [hunk 1]
```diff
@@ -63,7 +63,10 @@ Computation and Results
 
     Operations that introduce nontrivial mappings on indices (e.g. reshape,
     strided slice, roll) are identified as potential candidates for being captured
-    in their own high-level node vs. as an :class:`pytato.array.IndexLambda`.
+    in their own high-level node vs. as an :class:`~pytato.array.IndexLambda`.
+    
+    Operations that *can* be expressed as :class:`~pytato.array.IndexLambda`
+    without loss of information, *should* be expressed that way.
```
Comment (kaushikcfd) [inline doc/design.rst:69]: I don't think we've defined what is considered as "loss of information"?

Are math function to index lambdas not loss of information?

Comment (kaushikcfd) [reply doc/design.rst:69]: One reason for math functions have their own nodes is possibly one could perform some DAG traversals to perform algebraic simplifications.  (such as applying some trigonometric identities)

Comment (inducer) [reply doc/design.rst:69]: That's equally possible if they're in `IndexLambda`s.

Comment (inducer) [reply doc/design.rst:78]: > I don't think we've defined what is considered as "loss of information"?

Gave this a shot in c57ee7d.

Comment (kaushikcfd) [reply doc/design.rst:78]: Thank you, clear now.

**Comment by kaushikcfd** [comment]
LGTM from my end.

**Comment by inducer** [comment]
Thanks for taking a look!

**Pull request merged**